### PR TITLE
Bob pre key bundle

### PR DESCRIPTION
**Pull request by nealrea**
add schnorr values to session_pre_key_bundle struct.
add getters for session_pre_key_bundle schnorr values.
create new unit test for schnorr proof verification.

### CODE REVIEW

File: tests/test_session_builder.c [hunk 1]
```diff
@@ -65,6 +66,93 @@ void test_teardown()
...
+            bob_local_registration_id,
+            1, /* device ID */
+            31337, /* pre key ID */
+            ec_key_pair_get_public(bob_pre_key_pair),
+            22, /* signed pre key ID */
+            ec_key_pair_get_public(bob_signed_pre_key_pair),
+            signal_buffer_data(bob_signed_pre_key_signature),
+            signal_buffer_len(bob_signed_pre_key_signature),
+            ratchet_identity_key_pair_get_public(bob_identity_key_pair),
+            session_signed_pre_key_get_rhat(bob_signed_pre_key),
```
Comment (KirstynNatavio) [inline tests/test_session_builder.c:130]: I will be creating a function to generate Rhatfull

File: src/session_pre_key.c [hunk 2]
```diff
@@ -39,6 +39,10 @@ struct session_pre_key_bundle {
     ec_public_key *signed_pre_key_public;
     signal_buffer *signed_pre_key_signature;
     ec_public_key *identity_key;
+    const uint8_t *Rhatfull;
```
Comment (KirstynNatavio) [inline src/session_pre_key.c:42]: Should this be changed to be of type **ge_p3**

Comment (KirstynNatavio) [reply src/session_pre_key.c:42]: This is already resolved in another PR

**Review by KirstynNatavio — APPROVED** [review]
LGTM! I think we can already merge this to master.

**Review by jacinta-m — APPROVED** [review]

**Pull request merged**